### DISCUSSION TITLE
Argument of type '"postgres_changes"' is not assignable to parameter of type '"broadcast"'.

### DISCUSSION THREAD

**Discussion by andyjessop** [supabase/supabase#15630] [Questions] [answered]
I'm trying to setup a subscription to a channel with the following code (js client lib):

```ts
  const channel = client
    .channel('project-changes')
    .on(
      'postgres_changes', // <-- TS ERROR ON THIS LINE
      {
        event: '*',
        filter: `id=eq.${id}`,
        schema: 'public',
        table: 'project',
      },
      ({ record }: { record: Project }) => {
        signal.value = {
          data: record,
          error: null,
          loading: true,
          updating: false,
        };
      },
    )
    .subscribe();
```
I get the error:

```
No overload matches this call.
  The last overload gave the following error.
    Argument of type '"postgres_changes"' is not assignable to parameter of type '"broadcast"'.
```

Is this something to do with the shape of the payload? If so, what is the type of the payload? I can't find it in the docs here: https://supabase.com/docs/reference/javascript/subscribe

**Comment by GaryAustin1** [comment]
Not sure on error, but here is payload info: https://supabase.com/docs/guides/realtime/protocol#postgres-cdc-message

**Reply by andyjessop** [reply]
That's not correct, I'm afraid:

```
Property 'data' does not exist on type 'RealtimePostgresChangesPayload<{ [key: string]: any; }>'.
```

It doesn't seem like the type of the `payload` is anywhere in the docs. It would be great if someone from Supabase could comment.

**Reply by GaryAustin1** [reply, marked as answer]
You are right that is not it...
![image](https://github.com/supabase/supabase/assets/54564956/83825444-0de6-4cb3-926b-40bc1526054d)
That is the content of payload.
